sklearn/metrics/_regression.py

```diff
@@ -395,21 +395,29 @@ def mean_absolute_percentage_error(
     >>> mean_absolute_percentage_error(y_true, y_pred)
     112589990684262.48
     """
+    input_arrays = [y_true, y_pred, sample_weight, multioutput]
+    xp, _ = get_namespace(*input_arrays)
+
     y_type, y_true, y_pred, multioutput = _check_reg_targets(
         y_true, y_pred, multioutput
     )
     check_consistent_length(y_true, y_pred, sample_weight)
-    epsilon = np.finfo(np.float64).eps
-    mape = np.abs(y_pred - y_true) / np.maximum(np.abs(y_true), epsilon)
-    output_errors = np.average(mape, weights=sample_weight, axis=0)
+    epsilon = xp.asarray(xp.finfo(xp.float64).eps, dtype=xp.asarray(0.0).dtype)
+    y_true_abs = xp.asarray(xp.abs(y_true), dtype=xp.asarray(0.0).dtype)
+    mape = xp.asarray(xp.abs(y_pred - y_true), dtype=xp.asarray(0.0).dtype) / xp.where(
```

Comment:
I would rather use the floating point dtype used in the regressor's predictions than a device dependent dtype.


```suggestion
    epsilon = xp.asarray(xp.finfo(xp.float64).eps, dtype=y_pred.dtype)
    y_true_abs = xp.asarray(xp.abs(y_true), dtype=y_pred.dtype)
    mape = xp.asarray(xp.abs(y_pred - y_true), dtype=y_pred.dtype) / xp.where(
```

Reply:
We have a function defined `_find_matching_floating_dtype` in the array api utils. Maybe you could use that to get the float dtype once and then use it in all three places?

Reply:
Thank you @OmarManzoor I have made the changes accordingly